lib/statsd.js

```diff
@@ -513,12 +517,14 @@ exports.StatsD = Client;
  * Note that this will no-op with an early exit if the last socket create time
  * was too recent (within the UDS graceful restart rate limit).
  * @param client Client The statsd Client that may be getting a UDS error handler.
- * @param lastSocketCreateTime number The timestamp (in milliseconds since the
- *        epoch) when the current socket was created.
  */
-function udsErrorHandler(client, lastSocketCreateTime) {
+function udsErrorHandler(client, err) {
+  if (!err || !UDS_ERROR_CODES.includes(-err.code)) {
+    return;
+  }
+
   // recreate the socket, but only once within `udsGracefulRestartRateLimit`.
-  if (Date.now() - lastSocketCreateTime < client.udsGracefulRestartRateLimit) {
+  if (Date.now() - client.lastSocketCreateTime < client.udsGracefulRestartRateLimit) {
```

Comment:
Don't you need to account for code paths where `client.lastSocketCreateTime` is not set?